packages/client/src/components/Input/Input.tsx

```diff
@@ -0,0 +1,100 @@
+import styled from 'styled-components';
+import React, { useEffect, useState } from 'react';
+
+type InputProps = {
+  name: string;
+  type: string;
+  placeholder: string;
+  isValid?: boolean;
+  value?: string;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  width?: string;
+  backgroundColor?: string;
+  color?: string;
+  label?: string;
+  labelTextColor?: string;
+  validationTextColor?: string;
+  validationText?: string;
+};
+
+type ValidationStyledProps = Pick<InputProps, 'validationTextColor'>;
+
+type LabelStyledProps = Pick<InputProps, 'labelTextColor'>;
+
+type InputStyledProps = Pick<InputProps, 'width' | 'backgroundColor' | 'color'>;
+
+const Input = (props: InputProps) => {
+  const [state, setState] = useState('');
+  const [isTyping, setIsTyping] = useState(false);
+
+  useEffect(() => {
+    state ? setIsTyping(true) : setIsTyping(false);
+  }, [state]);
+
+  const handler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    setState(value);
+  };
+
+  return (
+    <DivStyled>
+      <LabelStyled>{isTyping ? props.placeholder : ''} </LabelStyled>
+      <InputStyled
+        {...props}
+        name={props.name}
+        type={props.type}
+        value={props.value}
+        placeholder={props.placeholder}
+        onChange={props.onChange ? props.onChange : handler}
+        backgroundColor={props.backgroundColor}
+        width={props.width}
+        color={props.color}
+      />
+      <ValidationStyled validationTextColor={props.validationTextColor}>
+        {props.isValid ? '' : props.validationText}
+      </ValidationStyled>
+    </DivStyled>
+  );
+};
+
+const DivStyled = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  width: fit-content;
+  margin: 0 0 5px 0;
+`;
+
+const ValidationStyled = styled.p<ValidationStyledProps>`
+  color: ${props => props.validationTextColor || 'red'};
```

Comment:
все цвета мы будем брать примерно так, пока еще четкой структуры нет, как и самих цветов, так что пускай пока отрабатывают фолбэки 'red' и другие запасные цвета
```suggestion
  color: ${({ theme }) => theme.input.validationTextColor || 'red'};
```

Reply:
Убрал, когда будут темы, добавлю.